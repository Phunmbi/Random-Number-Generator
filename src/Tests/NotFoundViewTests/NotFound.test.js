import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import NotFound from '../../Views/NotFoundView';

let wrapped;

beforeEach(() => {
  wrapped = mount (
    <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
      <NotFound/>
    </MemoryRouter>
  );
});

it('should take screenshot successfully', () => {
  expect(wrapped).toMatchSnapshot();
});
