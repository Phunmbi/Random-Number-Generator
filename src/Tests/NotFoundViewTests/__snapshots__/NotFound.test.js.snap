// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take screenshot successfully 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testkey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testkey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testkey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotFound>
      <div
        className="NotFound"
      >
        <div
          className="NotFound-header"
        >
          <h2>
            Error 404
          </h2>
          <p>
            We can't find the page you are currently looking for but we can help you find your way back 
            <Link
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                home
              </a>
            </Link>
          </p>
        </div>
      </div>
    </NotFound>
  </Router>
</MemoryRouter>
`;
